runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -251,6 +257,23 @@ MM_IndexableObjectAllocationModel::layoutDiscontiguousArraylet(MM_EnvironmentBas
 	if (NULL != spine) {
 		switch (_layout) {
 		case GC_ArrayletObjectModel::Discontiguous:
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+			if (extensions->indexableObjectModel.isDoubleMappingEnabled()) {
+				/**
+				 * There are some special cases where double mapping an arraylet is
+				 * not necessary; isArrayletDataDiscontiguous() details those cases.
+				 */
+				if (extensions->indexableObjectModel.isArrayletDataDiscontiguous(spine)) {
+					doubleMapArraylets(env, (J9Object *)spine);
+					/* 
+					 * If doublemap fails the caller must handle it appropriatly. In case
+					 * of JNI critical, if doublemap fails, it will fall back to copying
+					 * each element of the array to a temporary array. It might hurt performance
+					 * but execution won't halt.
+					 */
```

Comment:
This comment looks like in a wrong place. it should be close to place where `doublemap fails`